### PR TITLE
change gumbel softmax dimension

### PR DESCRIPTION
**Pull request by sdtblck**
somewhat certain we should be softmaxing over hdim here - pytorch's default is -1 which is the height dimension

### CODE REVIEW

**Comment by NaxAlpha** [comment]
-1 in this case is 1 because the shape of logits is `[batch_size, depth]` so -1 or 1 is in this case `depth`

**Comment by sdtblck** [comment]
@NaxAlpha is the shape not [b, hdim, h, w] ?

**Comment by NaxAlpha** [comment]
Ohh yeah! that is correct, I was using actually `get_codebook_indices` which has shape `[batch_size, seq_len]`

**Comment by NaxAlpha** [comment]
Wow! This is now working - I have been trying to figure out this issue with transformer convergence but with this fix, it seems it might actually work. Here are results without this fix (after a few hours):

image

But now hopefully, it ll converge 😀

**Comment by lucidrains** [comment]
Omg, thanks Sid! 😂🔥🔥
